symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/main/java/com/symphony/bdk/spring/config/BdkActivityConfig.java

```diff
@@ -88,7 +90,17 @@ public void setApplicationContext(final ApplicationContext applicationContext) t
             }
           }
         }
-      }
+      });
+    }
+
+    /**
+     * Returns a {@link Stream} of beans to be scanned from {@link ApplicationContext}.
+     * Only "singleton" beans will be returned.
+     */
+    private static Stream<Object> getBeansToScan(ApplicationContext applicationContext) {
+      return Arrays.stream(applicationContext.getBeanDefinitionNames())
```

Comment:
Don't still risk to create beans that we should not create?
Maybe lazy beans? 

I was trying to look at something similar (the application event) and the logic seems quite complex (org.springframework.context.event.EventListenerMethodProcessor#afterSingletonsInstantiated)

Reply:
Maybe a BeanPostProcessor could be another way to do it? https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-factory-extension-bpp

Reply:
@symphony-youri see latest commit https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/pull/347/commits/8b5fd0c18f9f5a2e1c1a6e446de7f42333037b52. I extracted the whole logic in a separate class, including what was done by the Spring `EventListenerMethodProcessor` class,  and based on the `BeanPostProcessor` interface as well that makes easier to handle lazy beans initialization.